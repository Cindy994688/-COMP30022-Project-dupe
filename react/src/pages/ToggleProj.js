import React, { Component } from 'react'
import Jumbotron from 'react-bootstrap/Jumbotron'
//thank you to https://www.youtube.com/watch?v=x5oiX93DeHA
import './pages.css'
import './darkmode.css'

export default class ToggleProj extends Component {
    state = {
        on: false,
    }

    toggle = () => {
        this.setState({
            on: !this.state.on
        })
    }
    render() {
        return (
            <div>
                <div className = {"topTriangle thirdColor"+this.props.colourMode}>
                    <div className={"intro"+this.props.colourMode + " title"+this.props.colourMode}> Our Project </div>
                    {/*<div className="title" id="ourproject">*/}
                    {/*<div className="title"> Our Project </div>*/}
                    <div className={"sub"+this.props.colourMode}> Bethâ€™s Beauty - Your best beauty services </div>
                    <div className={"buttondiv"+this.props.colourMode}>
                    <button type="submit" className={"btn"+this.props.colourMode+ " btn-primary"+this.props.colourMode+ " button"+this.props.colourMode} onClick = {this.toggle}  style={{cursor:'pointer'}}>learn about our project</button>
                    </div>
                    <div>&nbsp;&nbsp;</div>
                </div>
                {/*<button onClick = {this.toggle}>clickme</button>*/}

                {this.state.on && this.props.children}
            </div>
        )
    }
}
