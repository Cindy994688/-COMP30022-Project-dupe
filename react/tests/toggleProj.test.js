/**
 * @jest-environment jsdom
 */

const puppeteer = require('puppeteer');
import React from 'react';
import renderer from 'react-test-renderer';
import ToggleProj from '../src/pages/ToggleProj';


test('it renders correctly', () => {
  component = renderer.create(<ToggleProj/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('E2E: Main project section expands when clicked', async() => {
  const puppet = await puppeteer.launch({
    headless: true,
  });
  const page = await puppet.newPage();
  await page.goto(
    'http://localhost:3000/'
  );
  await page.click('div.groupPr');
  var response = await page.$('#projectIntro');
  response = response == null;
  await puppet.close();
  expect(response).toBe(false)
}, 100000);

test('E2E: Main project section dissapears when clicked again', async() => {
  const puppet = await puppeteer.launch({
    headless: true,
  });
  const page = await puppet.newPage();
  await page.goto(
    'http://localhost:3000/'
  );
  await page.click('div.groupPr');
  await page.click('div.groupPr');
  var response = await page.$('#projectIntro');
  response = response == null;
  await puppet.close();
  expect(response).toBe(true)
}, 10000);
