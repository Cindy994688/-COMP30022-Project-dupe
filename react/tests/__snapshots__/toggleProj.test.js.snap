// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    className="topTriangle thirdColorundefined"
  >
    <div
      className="introundefined titleundefined"
    >
       Our Project 
    </div>
    <div
      className="subundefined"
    >
       Beth’s Beauty - Your best beauty services 
    </div>
    <div
      className="buttondivundefined"
    >
      <button
        className="btnundefined btn-primaryundefined buttonundefined"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
        type="submit"
      >
        learn about our project
      </button>
    </div>
    <div>
        
    </div>
  </div>
</div>
`;
